### PR TITLE
fix: add minted coins to balance in x/collection MsgMintFT

### PR DESCRIPTION
**Pull request by 0Tech**
## Description

This patch would fix accumulation logic of balance in MsgMintFT.

closes: #1104

## Checklist:

- [x] I followed the [contributing guidelines](https://github.com/Finschia/finschia-sdk/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/Finschia/finschia-sdk/blob/main/CODE_OF_CONDUCT.md).
- [x] I have added a relevant changelog to `CHANGELOG.md`
- [x] I have added tests to cover my changes.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated API documentation `client/docs/swagger-ui/swagger.yaml`

### CODE REVIEW

File: x/collection/keeper/invariants.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	}
+}
+
+func TotalSupplyInvariant(k Keeper) sdk.Invariant {
+	return func(ctx sdk.Context) (string, bool) {
+		// cache, we don't want to write changes
+		ctx, _ = ctx.CacheContext()
+
+		invalidClassIDs := map[string][]string{}
+		k.iterateContracts(ctx, func(contract collection.Contract) (stop bool) {
```
Comment (zemyblue) [inline x/collection/keeper/invariants.go:28]: Doesn't it take a long time? Is it?

**Comment by 0Tech** [comment]
I've checked the migration working as expected.

Invariant relevant commits would be reverted as a result of an internal discussion.

File: x/collection/keeper/migrations/v2/store.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+		mintedKey := StatisticKey(MintedKeyPrefix, contractID, classID)
+		if !minted.IsZero() {
+			bz, err := minted.Marshal()
+			if err != nil {
+				return err
+			}
+			store.Set(mintedKey, bz)
+		} else {
+			store.Delete(mintedKey)
+		}
```
Comment (jaeseung-bae) [inline x/collection/keeper/migrations/v2/store.go:130]: I'm just wondering if it is okay to delete when minted value is zero.

File: x/collection/module/module.go [hunk 3]
```diff
@@ -107,6 +107,7 @@ func (am AppModule) LegacyQuerierHandler(legacyQuerierCdc *codec.LegacyAmino) sd
 func (am AppModule) RegisterServices(cfg module.Configurator) {
 	collection.RegisterMsgServer(cfg.MsgServer(), keeper.NewMsgServer(am.keeper))
 	collection.RegisterQueryServer(cfg.QueryServer(), keeper.NewQueryServer(am.keeper))
+	keeper.NewMigrator(am.keeper).Register(cfg.RegisterMigration)
```
Comment (zemyblue) [inline x/collection/module/module.go:110]: Don't need to handle error case?

File: CHANGELOG.md [hunk 4]
```diff
@@ -64,6 +64,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * (x/bank) [#1093](https://github.com/Finschia/finschia-sdk/pull/1093) Remove message events including `sender` attribute whose information is already present in the relevant events (backport #1066)
 * (ostracon) [\#1099](https://github.com/Finschia/finschia-sdk/pull/1099) feat!: remove libsodium vrf library.
 * (x/collection) [\#1102](https://github.com/finschia/finschia-sdk/pull/1102) Reject modifying NFT class with token index filled in MsgModify
+* (x/collection) [\#1105](https://github.com/finschia/finschia-sdk/pull/1105) Add minted coins to balance in x/collection MsgMintFT
```
Comment (zemyblue) [inline CHANGELOG.md:67]: Please change github org link to uppercase like `https://github.com/Finschia/`.

File: x/collection/keeper/migrations/v2/store.go (hunk 2 above)
Comment (0Tech) [reply x/collection/keeper/migrations/v2/store.go:130]: It's OK. Please refer: https://github.com/Finschia/finschia-sdk/blob/fe6883fc79d723f13bc1d54de938edd7c5ae08a3/x/collection/keeper/supply.go#L453-L465

File: x/collection/module/module.go (hunk 3 above)
Comment (0Tech) [reply x/collection/module/module.go:110]:
```suggestion
	if err := keeper.NewMigrator(am.keeper).Register(cfg.RegisterMigration); err != nil {
		panic(err)
	}
```

File: CHANGELOG.md (hunk 4 above)
Comment (0Tech) [reply CHANGELOG.md:67]:
```suggestion
* (x/collection) [\#1105](https://github.com/Finschia/finschia-sdk/pull/1105) Add minted coins to balance in x/collection MsgMintFT
```

**Review by jaeseung-bae — APPROVED** [review]

File: x/collection/keeper/invariants.go (hunk 1 above)
Comment (0Tech) [reply x/collection/keeper/invariants.go]: The relevant logic has been removed.

**Review by zemyblue — APPROVED** [review]

**Pull request merged**